lib/puppet/util/windows/registry.rb

```diff
@@ -31,16 +31,24 @@ def open(name, path, mode = KEY_READ | KEY64, &block)
     end
 
     def values(subkey)
+      orig_enc = Encoding.default_internal
+      Encoding.default_internal = Encoding::UTF_8
```

Comment:
For more context it errors in the `each_value` when it calls `export_string` because it is trying to convert to the system locale.

The package in question that can make it happen every time is `Windows Azure Libraries for .NET 1.7 â?" June 2012` (although I have somewhere between 4 and 6 that fail) without this change. Our handling code works fine on Ruby 1.9.3 and Ruby 2.0.0 because they don't call the export_string trying to handle encoding for us.